src/cmap/worker.rs

```diff
@@ -651,20 +636,19 @@ async fn establish_connection(
                     e.handshake_phase.clone(),
                 )
                 .await;
-            if let Some(handler) = event_handler {
-                let event = ConnectionClosedEvent {
+            event_emitter.emit_event(|| {
+                ConnectionClosedEvent {
                     address,
                     reason: ConnectionClosedReason::Error,
                     connection_id,
-                };
-                handler.handle_connection_closed_event(event);
-            }
+                    error: Some(e.cause.clone()),
```

Comment:
I just extracted the `cause` here since `EstablishError` is a different type than `Error`. AFAICT the extra info stored on the `EstablishError` is what phase of the handshake the error happened during. I wasn't sure if that info was worth including here or more exists so the driver can decide what to do about the error. LMK what you think.

Reply:
Yep, that's just implementation detail stuff for SDAM error handling purposes. Using `cause` here SGTM.